src/librustc/hir/mod.rs

```diff
@@ -1450,7 +1450,7 @@ pub struct ItemId {
 /// The name might be a dummy name in case of anonymous items
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct Item {
-    pub name: Name,
+    pub name: Spanned<Name>,
```

Comment:
Ah no worries, I just kept last changes so that it is easier to continue the discussion.

Okay so continue, when we display the error, we only have `hir::Item`. Specifically at this section

https://github.com/rust-lang/rust/blob/f1f40f850e2546c2c187514e3d61d17544ba433f/src/librustc/traits/error_reporting.rs#L571-L585

It seems that we can get the `hir::Item` by doing something like this `self.map.find(4)`, not sure if we can get `ast::Item` at that place?